File: CivitasJava/src/civitas/SorpresaConvertirJugador.java
package civitas;

import java.util.*;

public class SorpresaConvertirJugador extends Sorpresa{

    SorpresaConvertirJugador(String texto, int valor){
        super(texto);
        this.valor=valor;
    }

    @Override
    void aplicarAJugador(int actual, ArrayList<Jugador> todos){
        if(jugadorCorrecto(actual, todos)){
            informe(actual, todos);
            JugadorEspeculador nuevo = new JugadorEspeculador(todos.get(actual), valor);
            todos.set(actual, nuevo);
        }
    }
}

File: CivitasRuby/lib/sorpresa_convertir_jugador.rb
#encoding:utf-8

require_relative 'sorpresa'
require_relative 'jugador'
require_relative 'jugador_especulador'
require_relative 'casilla'
require_relative 'mazo_sorpresas'
require 'sorpresa.rb'

module Civitas
    class SorpresaConvertirJugador < Sorpresa 
        def initialize(valor, texto)
            super(texto)
            @valor=valor
        end

        def aplicar_a_jugador(actual,todos)
            if(jugador_correcto(actual,todos))
                informe(actual,todos)
                especulador=JugadorEspeculador.nuevo_especulador(todos[actual], @valor)
                todos[actual]=especulador
            end
        end

        public_class_method :new
    end
end